code/services/LDAPService.php

```diff
@@ -413,10 +414,50 @@ public function updateMemberFromLDAP($member, $data = null) {
 				Group::get()->byId($groupRecord['GroupID'])->Members()->remove($member);
 			}
 		}
-
 		// This will throw an exception if there are two distinct GUIDs with the same email address.
 		// We are happy with a raw 500 here at this stage.
 		$member->write();
 	}
 
+
+	/**
+	 * Change a members password on the AD. Works with ActiveDirectory compatible services that saves the
+	 * password in the `unicodePwd` attribute.
+	 *
+	 * Ensure that the LDAP bind:ed user can change passwords and that the connection is secure.
+	 *
+	 * @param Member $member
+	 * @param string $password
+	 * @return ValidationResult
+	 * @throws Exception
+	 */
+	public function setPassword(Member $member, $password) {
+		$validationResult = ValidationResult::create(true);
+		if (!$member->GUID) {
+			SS_Log::log(sprintf('Cannot update Member ID %s, GUID not set', $member->ID), SS_Log::WARN);
+			$validationResult->error(_t('LDAPAuthenticator.PROBLEMFINDINGDATA', 'There was a problem retrieving your user data'));
```

Comment:
"Problem retrieving data" is not really a validation failure - it would be much more consistent if we throw an exception here and below. We already have a try-catch in the caller, so that should just work. Another good reason is that the message "We couldn't accept that password: There was a problem retrieving your user data" doesn't make much sense.